ocs_ci/utility/networksplit/main.py

```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf8 -*-
+
+"""
+This module contains primary interface of network split module, expected to be
+used by test setup and teardown fixtures, or manually via command line tool.
+
+Until arbiter network split fixtures are implemented, API here is tentative.
+"""
+
+from datetime import datetime, timedelta
+import argparse
+import logging
+import os
+import sys
+
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.utility.networksplit import machineconfig
+from ocs_ci.utility.networksplit.zone import ZoneConfig
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.ocp import exec_oc_debug_cmd
+
+
+logger = logging.getLogger(name=__file__)
+
+
+VALID_NETWORK_SPLITS = ("ab", "bc", "ab-bc", "ab-ac")
+
+
+def _list_cluster_nodes(zone_name=None):
+    """
+    Get cluster nodes of a whole cluster or from given zone only.
+
+    Args:
+        zone_name (str): name of k8s topology zone to list nodes within, if not
+            specified, nodes from whole cluster will be listed
+
+    Returns:
+        list: node ip addressess (as strings)
+
+    """
+    oc_cmd = [
+        "oc",
+        "--kubeconfig",
+        os.getenv("KUBECONFIG"),
+        "get",
+        "nodes",
+        "-o",
+        "name",
+    ]
+    if zone_name is not None:
+        oc_cmd.extend(["-l", "topology.kubernetes.io/zone=" + zone_name])
+        logger.debug("trying to list nodes in %s zone", zone_name)
+    else:
+        logger.debug("trying to list all nodes")
+    nodes_str = run_cmd(cmd=oc_cmd, timeout=600)
+    return nodes_str.splitlines()
+
+
+def _get_all_node_ip_addrs(node):
+    """
+    Get all ip addresses (both internal and external) of given node.
+
+    Args:
+        node (str): name of OCP node
+
+    Returns:
+        list: node ip addressess (as strings)
+
+    """
+    ip_addrs = []
+    oc_cmd = ["oc", "--kubeconfig", os.getenv("KUBECONFIG"), "get", node, "-o", "yaml"]
+    logger.debug("trying to get details about %s", node)
+    node_str = run_cmd(cmd=oc_cmd, timeout=600)
+    node_dict = yaml.safe_load(node_str)
+    for addr_d in node_dict["status"]["addresses"]:
+        if addr_d["type"] not in ("ExternalIP", "InternalIP"):
+            continue
+        ip_addrs.append(addr_d["address"])
+    return ip_addrs
+
+
+def get_zone_config(zone_names=None):
+    """
+    Detect and create zone configuration based on ocs-ci config file, unless
+    zone_names are specified to override ocs-ci config.
+
+    Args:
+        zone_names (list): lislt of zone names
+
+    Returns:
+        ZoneConfig: object with list of node ip addresses for each zone
+
+    """
+    zn = {}
+    if zone_names is not None:
```

Comment:
Isn't enough to have just:
```suggestion
    if zone_names:
```
Also if you require to have len(zone_names) == 3 what about to add some assertion here to check for this length?
Worth to mention it also in the docstring for zone_names

Reply:
The if statement is there to check if zone names list was specified, and for that it works as expected. Using sheer `if zone_names` would be a bug. That said, I can check for length of the list and raise ValueError if the length is not 3, using sheer assert would also be a blatant bug.

Reply:
Please do at least documentation that it requires 3 zones and the exception sounds good, I meant something like that instead of fail with indexError in the case someone will pass less number of zones. If someone will pass more you will just ignore them so if you need only 3 there should be some check for that I think.

Reply:
resolved in separate project